src/smt/optimization_solver.cpp

```diff
@@ -46,92 +46,371 @@ OptResult OptimizationSolver::checkOpt()
   Assert(d_activatedObjective.getType() != OBJECTIVE_UNDEFINED);
   Assert(!d_activatedObjective.getNode().isNull());
 
+  std::unique_ptr<OMTOptimizer> optimizer = OMTOptimizer::getOptimizerForNode(
+      d_activatedObjective.getNode(), d_activatedObjective.getSigned());
+
+  Assert(optimizer != nullptr);
+
+  std::pair<OptResult, Node> optResult;
+  if (d_activatedObjective.getType() == OBJECTIVE_MAXIMIZE)
+  {
+    optResult = optimizer->maximize(this->d_parent,
+                                    this->d_activatedObjective.getNode());
+  }
+  else if (d_activatedObjective.getType() == OBJECTIVE_MINIMIZE)
+  {
+    optResult = optimizer->minimize(this->d_parent,
+                                    this->d_activatedObjective.getNode());
+  }
+
+  this->d_savedValue = optResult.second;
+  return optResult.first;
+}
+
+void OptimizationSolver::activateObj(const Node& obj,
+                                     const int& type,
+                                     bool bvSigned)
+{
+  d_activatedObjective = Objective(obj, (ObjectiveType)type, bvSigned);
+}
+
+Node OptimizationSolver::objectiveGetValue()
+{
+  Assert(!d_savedValue.isNull());
+  return d_savedValue;
+}
+
+Objective::Objective(Node obj, ObjectiveType type, bool bvSigned)
+    : d_type(type), d_node(obj), d_bvSigned(bvSigned)
+{
+}
+
+ObjectiveType Objective::getType() { return d_type; }
+
+Node Objective::getNode() { return d_node; }
+
+bool Objective::getSigned() { return d_bvSigned; }
+
+std::pair<OptResult, Node> OMTOptimizerInteger::optimize(
+    SmtEngine* parentSMTSolver, Node target, ObjectiveType objType)
+{
+  // linear search for integer goal
   // the smt engine to which we send intermediate queries
   // for the linear search.
   std::unique_ptr<SmtEngine> optChecker;
```

Comment:
why is this no longer a member?

Reply:
which one should be a member?

Reply:
`optChecker`. In current master, isn't it a member?

Reply:
it's in a different class...

Reply:
oh good question. the original integer optimizer creates the sub solver called optChecker and transfers the assertions from parental solver to optChecker. 
This might be wrong because the options are not transferred.

Reply:
don't know why they merged the original int optimizer to master then...

Reply:
A possibly correct one would be to directly use the parentalSMTSolver, with proper pushing and popping.

Reply:
oh initializeSubSolver will deal with it... My bad...